src/common/math/Vector.h

```diff
@@ -26,6 +26,16 @@ along with Daemon Source Code.  If not, see <http://www.gnu.org/licenses/>.
 #define COMMON_VECTOR_H_
 
 #include <ostream>
+#include <type_traits>
+
+#include "common/CPPStandard.h"
+
+// HACK: checking for CPP_INVOKE_RESULT doesn't work on MSVC
+#if __cplusplus >= 201703L
```

Comment:
In that case you would need to add back the compiler flag to set `__cplusplus`? But it needs to be set in the build system, not just the CI.

Instead of this maybe it would be easier to just make our own function. Like `template<typename Func, typename... Args> using resultof_t = decltype(std::declval<Func>()(std::declval<Args...>));` and then replace `std::result_of<Func(Type, Type2)>::type` with `resultof_t<Func, Type, Type2>>` etc.